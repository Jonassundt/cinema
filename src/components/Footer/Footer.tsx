import React, { useState } from 'react';

import styles from './footer.module.css';

function Footer() {
    const [credits, setCredits] = useState(false);
    const startCredits = () => {
        setCredits(true);
        setTimeout(() => window.scroll({
            top: 4000,
            behavior: 'smooth'
        }), 600)
        setTimeout(() => setCredits(false), 20000);
    }

    if (credits) {
        return (
            <div className={styles.CreditFooter} >
                <div className={styles.Fade}></div>
                <div className={styles.CreditText}>
                    <div className={styles.Crawl}>
                        <div className={styles.Title}>
                            <p>Episode II</p>
                            <h1>MODERNE HJEMMEKINO</h1>
                        </div>
                        <p>Prosjektet er utviklet av Team 60, og er ment for å være en web-basert filmplakat-
                        utstilling som en homage til noen av de beste filmene noensinne laget.</p>
                        <p>Dette valget ble tatt grunnet gruppemedlemmenes felles kjærlighet for gode filmer,
                        og et ønske om å hylle både komponistene og regissørene.</p>
                    </div>
                </div></div>)
    }
    return (
        /* Simple footer component.
        * Returns footer containing text and an image.
        */
        <div className={styles.Footer} onClick={startCredits}>
            <div className={styles.Row}>
                <div className={styles.About}>NTNU Datateknologi 2020</div>
                <div className={styles.Link}>Rulletekst {"--->"}</div>
            </div>
            <img src="icons/react.svg" alt="" width="60" height="60" />

        </div>

    );
}

export default Footer;
