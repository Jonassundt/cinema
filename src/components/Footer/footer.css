.Footer {
    padding: 40px;
    background-color: rgb(124, 124, 124, 0.5);
    border-radius: 20px;
    text-align: center;   
    margin-bottom: 100px; 
    padding: 100px;
}