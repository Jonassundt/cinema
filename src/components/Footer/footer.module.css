.Footer {
    background: #3333338e;
    border-radius: 20px;
    text-align: center;   
    padding: 80px;
    color: #cacaca;
    font-size: 1.2em;
}