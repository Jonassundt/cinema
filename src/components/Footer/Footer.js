import React from 'react';

import './footer.css';

function Footer() {
    return (
        <div className="Footer">
            TODO: Create a standard footer with information about the project
        </div>
    );
}

export default Footer;
