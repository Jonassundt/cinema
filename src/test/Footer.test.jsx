import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../components/Footer/Footer.tsx';

describe('testing the footer', () => {
    it("matches the snapshot", () => {
        const tree = renderer.create(<Footer />).toJSON();

        expect(tree).toMatchSnapshot();
    })
});