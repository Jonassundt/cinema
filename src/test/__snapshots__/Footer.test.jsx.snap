// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the footer matches the snapshot 1`] = `
<div
  className="Footer"
  onClick={[Function]}
>
  <div
    className="Row"
  >
    <div
      className="About"
    >
      NTNU Datateknologi 2020
    </div>
    <div
      className="Link"
    >
      Rulletekst 
      ---&gt;
    </div>
  </div>
  <img
    alt=""
    height="60"
    src="icons/react.svg"
    width="60"
  />
</div>
`;
